A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_AocRequestsWeather.js

```diff
@@ -0,0 +1,137 @@
+class CDUAocRequestsWeather {
+    static ShowPage(mcdu, store = { "reqType": 0, "depIcao": "", "arrIcao": "", "arpt1": "", "arpt2": "", "arpt3": "", "arpt4": "", "sendStatus": ""}) {
+        mcdu.clearDisplay();
+
+        if (mcdu.flightPlanManager.getOrigin() && mcdu.flightPlanManager.getDestination()) {
+            store.arpt1 = mcdu.flightPlanManager.getOrigin().ident;
+            store.arpt2 = mcdu.flightPlanManager.getDestination().ident;
+        }
+
+        const reqTypes = [
+            'METAR',
+            'TAF LONG',
+            'SIGMET',
+            'METAR + TAF'
+        ];
+
+        const updateView = () => {
+            mcdu.setTemplate([
+                ["AOC WEATHER REQUEST"],
+                [`WX TYPE`, "AIRPORTS"],
+                [`↓${reqTypes[0]}`, `${store.arpt1 != "" ? store.arpt1 : "[ ]"}[color]blue`],
+                [""],
+                ["", `${store["arpt2"] != "" ? store["arpt2"] : "[ ]"}[color]blue`],
+                [""],
+                ["", `${store["arpt3"] != "" ? store["arpt3"] : "[ ]"}[color]blue`],
+                [""],
+                ["", `${store["arpt4"] != "" ? store["arpt4"] : "[ ]"}[color]blue`],
+                [""],
+                [""],
+                ["RETURN TO", `${store["sendStatus"]}`],
+                ["<AOC MENU", "SEND*[color]blue"]
+            ]);
+        }
+        updateView();
+
+        mcdu.onRightInput[0] = () => {
+            let value = mcdu.inOut;
+            mcdu.clearUserInput();
+            store["arpt1"] = value;
+            CDUAocRequestsWeather.ShowPage(mcdu, store);
+        }
+
+        mcdu.onRightInput[1] = () => {
+            let value = mcdu.inOut;
+            mcdu.clearUserInput();
+            store["arpt2"] = value;
+            CDUAocRequestsWeather.ShowPage(mcdu, store);
+        }
+
+        mcdu.onRightInput[2] = () => {
+            let value = mcdu.inOut;
+            mcdu.clearUserInput();
+            store["arpt3"] = value;
+            CDUAocRequestsWeather.ShowPage(mcdu, store);
+        }
+
+        mcdu.onRightInput[3] = () => {
+            let value = mcdu.inOut;
+            mcdu.clearUserInput();
+            store["arpt4"] = value;
+            CDUAocRequestsWeather.ShowPage(mcdu, store);
+        }
+
+        mcdu.onRightInput[5] = async () => {
+            store["sendStatus"] = "QUEUED"
+            updateView();
+            const ICAOS = [store["arpt1"], store["arpt2"], store["arpt3"], store["arpt4"]];
+            mcdu.clearUserInput();
+            const lines = []; // Prev Messages
+            let errors = 0;
+            const getData = async () => {
+                for (const icao of ICAOS) {
+                    if (icao !== "") {
+                        await fetch(`https://us-central1-flybywire-metar.cloudfunctions.net/metar-py?source=ms&icao=${icao}`)
```

Comment:
You could speed this up in the case of multiple requests by doing something like:
```js
const parts = await Promise.all(ICAOS
  .filter((icao) => icao !== '')
  .map(async (icao) => {
    const data = await fetch('...').then((r) => r.text());
    if (data.slice(0, 9) === 'FBW_ERROR') {
      return [
        // ...
      ];
    }
    return [
      `METAR ${icao}`,
      ...wordWrapToStringList(data, 25),
      '------------------------',
    ];
  });
const lines = parts.flat();